networkx/utils/backends.py

```diff
@@ -811,14 +811,18 @@ def __call__(self, /, *args, backend=None, **kwargs):
             if len(backend_names) != 1:
                 # Future work: convert between backends and run if multiple backends found
                 raise TypeError(
-                    f"{self.name}() graphs must all be from the same backend, found {backend_names}"
+                    f"{self.name}() graphs must all be from the same backend, found "
+                    f"{backend_names}. This may be supported via backend-to-backend "
+                    "conversions in the future."
```

Comment:
Minor nit, nothing to block my approval: I think the original message was more succinct, and IMO mentioning backend-to-backend conversions here seems distracting. Perhaps instead is there some text that can be added to help the user resolve this problem if they were to get this error?

Reply:
By including this in the text, it serves as a kind of warning or advertisement that it may be implemented in the future. Explaining _how_ `NotImplementedError` behavior may be implemented seems like good practice to me--one that I would like to see more of.

How would you suggest users resolve this problem if they get this error?

Reply:
Interestingly, I also added a blurb to these messages in #7496. Do you like its verbiage better?
> "Automatic conversions between backends may become possible in future releases."